### PR TITLE
Scatter and Pythia Updates

### PR DESCRIPTION
**Pull request by vkbo**
This PR updates the PYTHIA interface to work with version 8.243. In one of the last versions, Pythia added the ability to send two particles, in terms of their 3-momenta, to the event generator. This is taken advantage of in the profile code to scatter the actual tracked particle against a Monte Carlo sample of beam 2.

### CODE REVIEW

File: doc/user_manual/chSpecialElements.tex [hunk 1]
```diff
@@ -1509,34 +1526,94 @@ \section{Scattering} \label{sec:scatter}
...
-This field can be controlled through DYNK, for example in order to scale only at one specified turn, or to switch off and on scattering at the element by scaling it by 0 and 1 respectively.
-The \texttt{PRO}file, \texttt{GEN}erator(s), and single elements are referenced through their names, and for the \texttt{GEN}erators and \texttt{PRO}file they must be defined above the \texttt{ELEM}ent in the \texttt{SCATTER} block.
+If the \texttt{PYTHOIA} block has the \texttt{REALBEAM} flag enabled (see Section~\ref{ExtPythia}), the Scatter Module will generate a random three-momentum vector of beam 2 particles that is then fed to the PYTHIA event generator.
+This flag enables the logging of these generated events to the \texttt{scatter\_momentum.dat} file.
+
+\paragraph{Density Dump}~ \texttt{DENSITY\_DUMP element\_name turn}\\
+
+This setting allows for a dump of the target PDF in a 500 by 500 matrix to the file \texttt{scatter\_pdf.dat}, and a dump of the target density seen by each particle to the file \texttt{scatter\_density.dat}.
+This is intended as a debug tool to check that beam 1 and beam 2 is actually interacting.
```
Comment (rdemaria) [inline doc/user_manual/chSpecialElements.tex]: Beam 1 and Beam 2

File: doc/user_manual/chSpecialElements.tex [hunk 2]
```diff
@@ -1509,34 +1526,94 @@ \section{Scattering} \label{sec:scatter}
...
-The element must be of type 40, as described in Section~\ref{SCAT}.
-The \texttt{ratio} argument (string/float) can be used to set a fixed scattering probability at the element.
-Alternatively, the field can be set to ``auto'', in which case the scattering probability is calculated from the cross section and density profile.
-The \texttt{scaling} argument (float) is used to scale the probability of an interaction when using cross section to claculate the scattering probability.
-This field can be controlled through DYNK, for example in order to scale only at one specified turn, or to switch off and on scattering at the element by scaling it by 0 and 1 respectively.
-The \texttt{PRO}file, \texttt{GEN}erator(s), and single elements are referenced through their names, and for the \texttt{GEN}erators and \texttt{PRO}file they must be defined above the \texttt{ELEM}ent in the \texttt{SCATTER} block.
+If the \texttt{PYTHOIA} block has the \texttt{REALBEAM} flag enabled (see Section~\ref{ExtPythia}), the Scatter Module will generate a random three-momentum vector of beam 2 particles that is then fed to the PYTHIA event generator.
```
Comment (rdemaria) [inline doc/user_manual/chSpecialElements.tex]: Beam 2

File: doc/user_manual/chExternal.tex [hunk 3]
```diff
@@ -6,18 +6,90 @@ \chapter{External Tools} \label{ExtTools}
...
+If the \texttt{MODES} are set to \texttt{MANUAL}, the cross sections can be specified as a second argument.
+
+\bigskip
+\noindent Available process types are:
+\begin{itemize}
+  \item Elastic: \texttt{EL}, \texttt{ELASTIC}
+  \item Single diffractive: \texttt{SD}, \texttt{SINGLEDIFF},\texttt{SINGLEDIFFRACTIVE}
+  \item Double diffractive: \texttt{DD}, \texttt{DOUBLEDIFF},\texttt{DOUBLEDIFFRACTIVE}
+  \item Central diffractive: \texttt{CD}, \texttt{CENTRALDIFF},\texttt{CENTRALDIFFRACTIVE}
+  \item Non-diffractive: \texttt{ND}, \texttt{NONDIFF},\texttt{NONDIFFRACTIVE}
```
[the remark this answers is not in the document]
Comment (vkbo) [reply doc/user_manual/chExternal.tex]: Yes, that is what it means. This is the terminology used in the PYTHIA manual, so I just want to be consistent.

File: source/mod_pythia.f90 [hunk 4]
```diff
@@ -382,11 +525,14 @@ subroutine pythia_postInput
...
 
-  call pythia_getCrossSection(sigmaTot, sigmaEl)
-  write(lout,"(a)")       "    Cross Sections"
-  write(lout,"(a,f12.6)") "    * Total:   ",sigmaTot
-  write(lout,"(a,f12.6)") "    * Elastic: ",sigmaEl
+  call pythia_getInitial(sigmaTot, sigmaEl, m0b1, m0b2)
+  pythia_partMass(1) = real(m0b1, kind=fPrec)*c1e3
+  pythia_partMass(2) = real(m0b2, kind=fPrec)*c1e3
+  write(lout,"(a)")       "PYTHIA> Other Settings"
+  write(lout,"(a,f12.6)") "PYTHIA>  * Pythia Mass:   ",pythia_partMass(1)
```
[the remark this answers is not in the document]
Comment (vkbo) [reply source/mod_pythia.f90]: This is a check to ensure that the mass of Beam 1 on SixTrack and PYTHIA are the same, this is not relevant for Beam 2.

File: doc/user_manual/chExternal.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (amereghe) [reply doc/user_manual/chExternal.tex]: Ok but I would also be explicit in the manual, for people maybe new to PITHYA

File: source/mod_pythia.f90 (hunk 4 above)
[the remark this answers is not in the document]
Comment (amereghe) [reply source/mod_pythia.f90]: of course, we don't track the other beam in 6T, but maybe an echo can be useful for crosschecks

**Pull request merged**